src/app.jsx

```diff
@@ -38,11 +38,49 @@ export const Application = () => {
             setCurrentUser(user.name || "");
         })
                 .then(() => {
+                    let result = [];
                     const currentPath = path.join("/");
-                    cockpit.spawn(["ls", "-p", `/home/${currentUser}/${currentPath}`], { superuser: true }).then((res) => {
-                        res = res.split("\n");
-                        setFiles(res.slice(0, res.length - 1));
+                    const channel = cockpit.channel({
+                        payload: "fslist1",
+                        path: `/home/${currentUser}/${currentPath}`,
+                        superuser: "try",
+                        watch: true,
                     });
```

Comment:
Please use [cockpit.file().watch()](https://cockpit-project.org/guide/latest/cockpit-file.html#cockpit-file-notify) for this. That's the same channel type under the hood, but  looks nicer.

I know it's pre-existing code, but this `path` looks wrong. Is navigator really meant to *only* browse the user's home directory, not any system dir? Also, it is not at all a given that home directories are in `/home/${username}` -- our own "Accounts" UI allows you to put them somewhere else. This needs to come from [cockpit.user().home](https://cockpit-project.org/guide/latest/cockpit-login.html).

If the restriction to the user's home directory is intended, then please drop the `superuser`. It's way too dangerous and also unnecessary to do anything as root there.

Reply:
This was fixed in the previous PR, I will rebase this onto `main`